### PR TITLE
cmds: Add a unittest for `cmd_parse_putval`.

### PR DESCRIPTION
**Pull request by octo**
While looking into #4119, I wanted to ensure that the provided timestamp is parsed correctly.

### CODE REVIEW

File: src/utils/cmds/putval.c [hunk 1]
```diff
@@ -187,22 +185,30 @@ cmd_status_t cmd_parse_putval(size_t argc, char **argv,
...
-    status = cmd_parse_option(argv[i], &key, &value, err);
+    status = cmd_parse_option(option, &key, &value, err);
     if (status == CMD_OK) {
       int option_err;
 
       assert(key != NULL);
       assert(value != NULL);
       option_err = set_option(&vl, key, value, err);
+      key = NULL;
+      value = NULL;
```
Comment (eero-t) [inline src/utils/cmds/putval.c:201]: Why?  These are are internal to loop, set to NULL at start of loop, and not used before looping again?

**Review by eero-t — DISMISSED** [review]
Looks fine, but I wonder about added NULLs (they do not harm anything so setting this already approved).

**Comment by eero-t** [comment]
Some distros do not like the code, and tests fail:
```
 make[2]: Entering directory '/__w/collectd/collectd'
  CC       src/utils/common/common_test.o
In file included from /usr/include/string.h:495,
                 from ./src/daemon/collectd.h:51,
                 from ./src/utils/common/common.h:31,
                 from src/utils/common/common_test.c:32:
In function 'strncpy',
    inlined from 'test_parse_values' at src/utils/common/common_test.c:316:5:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:10: error: '__builtin_strncpy' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/utils/common/common_test.c: In function 'test_parse_values':
src/utils/common/common_test.c:314:21: note: length computed here
  314 |     size_t buf_sz = strlen(cases[i].buffer) + 1;
      |                     ^~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

Apparently their compilers do not notice that dst buffer size depends src buffer size too...

**Comment by octo** [comment]
🤬

```
../../src/utils/common/common_test.c: In function 'test_parse_values':
../../src/utils/common/common_test.c:316:5: error: 'strncpy' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
  316 |     strncpy(buf, cases[i].buffer, sizeof(buf));
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/utils/common/common_test.c:314:21: note: length computed here
  314 |     size_t buf_sz = strlen(cases[i].buffer) + 1;
      |                     ^~~~~~~~~~~~~~~~~~~~~~~
```

**Review by eero-t — APPROVED** [review]

**Pull request merged**